modules/cluster/addons/aws-load-balancer-controller.yaml

```diff
@@ -26,9 +26,9 @@ metadata:
+  tls.key: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFb2dJQkFBS0NBUUVBdlZIeEZYa3E3NFdHZ2QzdTZ0WVZmVm5KVWs2RGUwWmhhbm94MFd2eVRFZHk3WWhiClNLcnpITTlhTVNIREdmQmh6NUZHQmdVMkFMUHVXVUZMUXo4RUNNTWp6TjJOY1I0cEVnVVU5OTdsT2wwWXZISDAKWjhjOEpVRm5JVGtzRlprdFNSUnN6M2R2TS8zbVVuMzUvdVFtUlFmdmxPZ0VnanVmZkR2OU8rc3FjT01VU01PZgp1KzBuanNtSS93ckhzdEJPWFE4aXJZQmlzYW83MFA4MnhLMmQ3cGIwMUtxa3FkQVVvR3R0MVYzOWd2cnpFVzBWClZ3MHpjeHhVTHF2NExrd2Nubm05MDhXaE5iaEVNK25pTUM4Y2psb1FPcHZuTnI0bnJHRWZFZldYM3R1UkFPdVIKc29xNWtDSXVBZGZLcGgra3Q4U05iQUJIc3R6Z0tyN2tpb3MxelFJREFRQUJBb0lCQUROeUhmSTQzL3NBck9HRApmcFlnSU96VVJrbndmZmcwMHVvTlZKWlZJQjJzV01YRmhSREZLNVZMbjNXMGVBRUhlbnplUEVvTWhuVkFiK2VmClA2V0JiVVk0U1BVMXkyYk96RmhTT3pWMFE1Uk5raGc3V3pBTVN3TFF1QWFtTTUyVzUyQnhoWFFIUDVmQVNpTm0KaWxYQ2g4ZkFkVkZnajBFWWxLazgzLzYxVSs0aXZweE9nNVB2S280T2VrSzErN3prdVR6c2NpZG5vLytBc293Wgo4a1JXYU1hTWlacEhFT09SMzl5aml0VitUeGhTcEEyQVZpTDliSjBPSmFtQ0dqOGxnWnpkNTlWWTQrZCszbEUrClVBTlUwN3RTWXZsMkpMVzY1K3h1YXp5eW5Bd1hKeVYxVVpGYmJEUUZhRCtTOWUxck9sL0d4Z1dpbllUR3ZlUWMKdlZwSVNJRUNnWUVBOVI5UTdKVTE2NW9VNktFU2NsVnBDeU1BWmpROXlFRStTc2c0ZTlQb0UzUXNXdk1sK0ZJaAovN2RyNUF3SkxudXJFZUNqa3draVZ2cm9LazVSWUFnRm8xRkxmRHJISkNNY0NtdXJ0Sm13QmVuUUZKTTVocTVPCkZ5V2ZHUEZZaFFZamZsZE95OHhHZ1hDSWNDbHVPRll5YU91YkZ0aFQrMW1SSlNlNUlMdlhadTBDZ1lFQXhiaXcKMEpXVDJkU2lrVEx4WWszTTFTM0IzNUNKTFhLY2Faa21NN285RFE4ZVpmbnlHeGRCZVdXTXRMVjkrRFFCeHJYUgo1UW8wU3d3eC96N3R4RDluNmJpWWhNdkZDeDVVTmZ4VDg3KzlhcDR2QSt2b05CM2x5TTVQZlp2U1lQMzJEYTRhCjRlUk1HbGpSVXl1cS80cGZTUzRxdXljbUc1SFQ5di92Ny9PMVRtRUNnWUJpdXZ0Z1gzVTVjWFZzZ2J3cFd5ZVYKT2NMbVo3SmUzNUNVbEhMVHhYV2JxRlpaUjhpS2JhS09HallWdzVkTEpycXYxN2hxOStuaCtPdmNabkUyYWVURApkSElnRnBpeGdJYkIxNzV3WHRyY2R1Rzk5OUxhNFloSW9IbWxsVWhmUkNxMXI3cm9NVjVRc3AveDlXNFV1UHRTCjhWc2xaM3JrS1o1aHhjODJWUktjdlFLQmdDdkRoNCtEckNncEFwcTA5ZXpXWDZKSDhJTzFrRWQ0K05uK3hnL3AKQng1ZTNGR2lFMDZmeFNSRHIvTkh3TXd4WnJzKy8yaDg2N280eHRCQ2M0cGVwbmQ3SG5jMlN6VDh6NUxyZVFHWQp6U3BPUFFYWEw1eHBpYlQ2eFdsK0ZnRFY5YUp0aVJHNjRJVnNaMWpMYjV3RVdyYnQreE5WeFArUk5hT1VYdDF6CnlndkJBb0dBTlJsRUR6SkhtZjJoNW1sN3Qra1kyckxvSjNEWnRDZlNaNlQwaCtxWDFjNmpyWVBtbjJ2QkJqOGwKR1FLQTd5RVNXN1J0SWY5VDB4V2ZEeTFLZVJYWHVwMzZoNkNrd3RVdlZwY1d6MWYwSVBlbjNINWdHSnBoVGNxWgo4bHM1VER5bU5FU2V1WEhFQW10MlIzNDIvelhIekpYVGtKU1A4d3RCUzlMZUhJRmlsUFU9Ci0tLS0tRU5EIFJTQSBQUklWQVRFIEtFWS0tLS0tCg==
```

Comment:
hmm, I'm kinda wary about this...

Reply:
It feels wrong to have certificate credentials as part of a generic deploy. Perhaps these should be tf vars or something?

Reply:
Yeah it doesn't feel nice - @dan-slinky-ckpd and I had a chat about this a couple of weeks ago.
It's ended up in here because the current process for including helm charts in the module is to template the chart [here](https://github.com/cookpad/terraform-aws-eks/blob/master/hack/generate_addons.sh#L14) then include the manifests. This chart generates the cert when you template https://github.com/aws/eks-charts/blob/0db3d7d464ddf8f0d7fcae8d5c77d5f7276ab216/stable/aws-load-balancer-controller/templates/webhook.yaml#L1

Some discussion on the project https://github.com/aws/eks-charts/issues/347

I guess there was a design choice in the way that we include helm add-ons (template + commit generated manifests) - easier to debug, don't require users to install helm 🤔

I think the solution would be to add a new command to combine the template+deploy for helm charts and use that [here](https://github.com/cookpad/terraform-aws-eks/blob/master/modules/cluster/addons.tf#L136).
For now though - this cert isn't used anywhere I think, we're terminating SSL at the load balancer - I believe the cert is using for integrating with cert manager, but we're not using cert manager with the load balancer controller.

Reply:
We can avoid the cert generation from the helm templating if we use cert-manager https://github.com/fernando-villalba/aws-load-balancer-controller/commit/668af05a45bace8457e0acce55ad2eec81abbfa5 - I think this is the cleanest solution

Reply:
Discussed in person - will handle in a separate PR from issue https://github.com/cookpad/terraform-aws-eks/issues/205